File: viewer/src/main/scala/farjs/viewer/ViewerSearch.scala
package farjs.viewer

import farjs.filelist.theme.FileListTheme
import farjs.ui.popup.{StatusPopup, StatusPopupProps}
import scommons.react._
import scommons.react.hooks._

import scala.scalajs.js

case class ViewerSearchProps(searchTerm: String,
                             onComplete: () => Unit)

object ViewerSearch extends FunctionComponent[ViewerSearchProps] {
  
  private[viewer] var statusPopupComp: ReactClass = StatusPopup

  protected def render(compProps: Props): ReactElement = {
    FileListTheme.useTheme
    val props = compProps.wrapped
    
    useLayoutEffect({ () =>
      //TODO: start search
      ()
    }, Nil)

    <(statusPopupComp)(^.plain := StatusPopupProps(
      text = s"""Searching for\n"${props.searchTerm}"""",
      title = "Search",
      onClose = { () =>
        // stop search
        props.onComplete()
        //inProgress.current = false
      }: js.Function0[Unit]
    ))()
  }
}

File: viewer/ViewerSearch.mjs
import React, { useLayoutEffect } from "react";
import StatusPopup from "@farjs/ui/popup/StatusPopup.mjs";

const h = React.createElement;

/**
 * @typedef {{
 *  readonly searchTerm: string;
 *  onComplete(): void;
 * }} ViewerSearchProps
 */

/**
 * @param {ViewerSearchProps} props
 */
const ViewerSearch = (props) => {
  const { statusPopupComp } = ViewerSearch;

  useLayoutEffect(() => {
    //TODO: start search
  }, []);

  return h(statusPopupComp, {
    text: `Searching for\n"${props.searchTerm}"`,
    title: "Search",
    onClose: () => {
      // stop search
      props.onComplete();
      //inProgress.current = false
    },
  });
};

ViewerSearch.displayName = "ViewerSearch";
ViewerSearch.statusPopupComp = StatusPopup;

export default ViewerSearch;
